digest/digest.go

```diff
@@ -123,6 +123,20 @@ func (d Digest) Algorithm() Algorithm {
 	return Algorithm(d[:d.sepIndex()])
 }
 
+// Verifier returns a writer object that can be used to verify a stream of
+// content against the digest. If the digest is invalid, an error will be
+// returned.
+func (d Digest) Verifier() (Verifier, error) {
+	if err := d.Validate(); err != nil {
```

Comment:
Probably not. Let's check the cases:

```
"" - verification will fail
"sha256:notenough" - always fails
```

I guess since the calculated hash is always valid, we will never return true for junk data.